src/odemis/gui/cont/acquisition.py

```diff
@@ -449,6 +453,432 @@ def open_acquisition_dialog(self):
+            )
+            return
+        # save the data
+        executor = futures.ThreadPoolExecutor(max_workers=2)
+        scheduled_future = executor.submit(self._export_data, future)
+        scheduled_future.add_done_callback(self._on_export_data_done)
+
+    def _reset_acquisition_gui(self, text=None, state=None):
+        """
+        Resets some GUI widgets for the next acquisition
+        text (None or str): a (error) message to display instead of the
+          estimated acquisition time
+        state (str): the state of the acquisition future
+        """
+        # hide/enable some widgets
+        self._panel.gauge_cryosecom_acq.Hide()
+        self._panel.btn_cryosecom_acqui_cancel.Hide()
+        self._panel.btn_cryosecom_acquire.Enable()
+        self._panel.btn_cryosecom_change_file.Enable()
+        self._panel.btn_acquire_overview.Enable()
+        self._panel.z_stack_chkbox.Enable()
+        self._panel.streams_chk_list.Enable()
+        self._panel.txt_cryosecom_left_time.Hide()
+        self._tab.streambar_controller.resume()
+        self._panel.Layout()
+
+        if state == ST_FINISHED:
+            # update the counter of the filename
+            self._config.fn_count = update_counter(self._config.fn_count)
+            # reset the storing directory to the project directory
+            self._config.last_path = self._config.pj_last_path
+        elif state == ST_FAILED:
+            self._panel.txt_cryosecom_est_time.Show()
+            self._panel.txt_cryosecom_est_time.SetLabel(text)
+        elif state == ST_CANCELED:
+            self._panel.txt_cryosecom_est_time.Show()
+            self._update_acquisition_time()
+
+    def _display_acquired_data(self, data):
+        """
+        Shows the acquired image on the view
+        """
+        # get the localization tab
+        local_tab = self._tab_data.main.getTabByName("cryosecom-localization")
+        local_tab.display_acquired_data(data)
+
+    @call_in_wx_main
+    def _on_export_data_done(self, future):
+        """
+        Called after exporting the data
+        """
+        self._reset_acquisition_gui(state=ST_FINISHED)
+        self._update_acquisition_time()
+        data = future.result()
+        self._display_acquired_data(data)
+
+    def _export_data(self, future):
+        """
+        Called to get the data from the future,
+        and to export the acquired data
+        """
+        st = stream.StreamTree(streams=list(self._acquiStreams.value))
+        thumb = acqmng.computeThumbnail(st, future)
+        data, _ = future.result()
+        filename = self._filename.value
+        if data:
+            exporter = dataio.get_converter(self._config.last_format)
+            exporter.export(filename, data, thumb)
+            logging.info(u"Acquisition saved as file '%s'.", filename)
+            # update the filename
+            self._filename.value = create_filename(
+                self._config.last_path,
```

Comment:
The filename should be generated the same way as in the model: with `pj_last_path`.